### PR TITLE
ComputerVision: Fix arguments for /analyze and /describe

### PR DESCRIPTION
**Pull request by cthrash**
Updates for [PR1](https://github.com/Azure/azure-rest-api-specs/pull/3479) and [PR2](https://github.com/Azure/azure-rest-api-specs/pull/3486).

* The API client name was changed from ComputerVisionAPI to ComputerVisionClient, in keeping with other Azure SDKs.
* The way the Azure region is specfied has changed.  Specifically, the AzureRegion property was dropped in favor of an Endpoint property.  If you were previously specifying an AzureRegion value, you should now specify Endpoint='https://{AzureRegion}.api.cognitive.microsoft.com' instead. This change ensures better global coverage.
* Some inconsistencies in argument types were corrected.  (a) maxCandidate is an integer for DescribeImage. (b) details is a array of enumerated types for AnalyzeImage. (c) isBWImg in the AnalyzeImage(color) response is a non-nullable bool. (d) clipArtType/lineDrawing in AnalyzeImage(imageType) is an integer (index) instead of a double.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/_metadata/cognitiveservices_data-plane_ComputerVision.txt [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
 1) azure-rest-api-specs repository information
 GitHub fork: Azure
 Branch:      master
-Commit:      9754d04e95bc4ffe16186aa76f652773923f5e10
+Commit:      1a8570aeb1c216495ab782ec4e19db30ee2cacbf
 
 2) AutoRest information
 Requested version: latest
-Bootstrapper version:    autorest@2.0.4282
+Bootstrapper version:    C:\Users\cthrash.REDMOND\AppData\Roaming\npm  `-- autorest@2.0.4283
```
Comment (dsgouda) [inline src/SDKs/_metadata/cognitiveservices_data-plane_ComputerVision.txt]: Please pull latest changes from psSdkJson6, run msbuild build.proj and regenerate the code

File: src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Generated/Models/ImageType.cs [hunk 2]
```diff
@@ -50,13 +50,13 @@ public ImageType(double clipArtType = default(double), double lineDrawingType =
         /// Gets or sets confidence level that the image is a clip art.
         /// </summary>
         [JsonProperty(PropertyName = "clipArtType")]
-        public double ClipArtType { get; set; }
+        public int ClipArtType { get; set; }
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Generated/Models/ImageType.cs]: Was this already an incorrect SDK which is being fixed or has something at the service end changed? Applies to other parameters/members too

File: src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Microsoft.Azure.CognitiveServices.Vision.ComputerVision.csproj [hunk 3]
```diff
@@ -9,12 +9,14 @@
...
-    2) Support customizing service endpoints by assigning the endpoint string to ComputerVisionClient.Endpoint. The endpoint string can be found on Azure Portal, it should contain only protocol and hostname, for example: https://westus.api.cognitive.microsoft.com
+    1) The API client name was changed from ComputerVisionAPI to ComputerVisionClient, in keeping with other Azure SDKs.
+    2) The way the Azure region is specfied has changed.  Specifically, the AzureRegion property was dropped in favor of an Endpoint property.  If you were previously specifying an AzureRegion value, you should now specify Endpoint='https://{AzureRegion}.api.cognitive.microsoft.com' instead. This change ensures better global coverage.
+    3) Some inconsistencies in argument types were corrected.  (a) maxCandidate is an integer for DescribeImage. (b) details is a array of enumerated types for AnalyzeImage. (c) isBWImg in the AnalyzeImage(color) response is a non-nullable bool. (d) clipArtType/lineDrawing in AnalyzeImage(imageType) is an integer (index) instead of a double.
     ]]>
-    </PackageReleaseNotes>
+</PackageReleaseNotes>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Microsoft.Azure.CognitiveServices.Vision.ComputerVision.csproj]: nit: indentation

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Generated/Models/ImageType.cs (hunk 2 above)
Comment (cthrash) [reply src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Generated/Models/ImageType.cs]: The service hasn't changed, we were preparing to make a version bump due to the name change (ComputerVisionAPI->ComputerVisionClient) so we're piling on with other changes we'd intended to fix.  The service has not changed in a while.

**Comment by cthrash** [comment]
@yangyuan fyi

**Review by yangyuan — APPROVED** [review]

File: src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Cognitive Services ComputerVision SDK")]
 [assembly: AssemblyDescription("Provides access to the Microsoft Cognitive Services ComputerVision APIs.")]
 
-[assembly: AssemblyVersion("3.0.0.0")]
-[assembly: AssemblyFileVersion("3.1.0.0")]
+[assembly: AssemblyVersion("3.2.0.0")]
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Properties/AssemblyInfo.cs]: `AssemblyVersion` remains the same, `AssemblyFileVersion` must be updated

File: src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Microsoft.Azure.CognitiveServices.Vision.ComputerVision.csproj [hunk 5]
```diff
@@ -9,10 +9,12 @@
     <Version>3.1.0-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Microsoft.Azure.CognitiveServices.Vision.ComputerVision.csproj]: Please bump the version number here.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great apart from a couple of minor comments.

File: src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (cthrash) [reply src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Properties/AssemblyInfo.cs]: Remains the same meaning it shouldn't be changed?

Comment (dsgouda) [reply src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Properties/AssemblyInfo.cs]: Yes, `AssemblyVersion` must not be changed

File: src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Microsoft.Azure.CognitiveServices.Vision.ComputerVision.csproj [hunk 6]
```diff
@@ -6,13 +6,15 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.CognitiveServices.Vision.ComputerVision</PackageId>
     <Description>This client library provides access to the Microsoft Cognitive Services ComputerVision APIs.</Description>
-    <Version>3.1.0-preview</Version>
+    <Version>3.2.0</Version>
     <AssemblyName>Microsoft.Azure.CognitiveServices.Vision.ComputerVision</AssemblyName>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Vision/ComputerVision/ComputerVision/Microsoft.Azure.CognitiveServices.Vision.ComputerVision.csproj]: Please acknowledge that this is a stable release. Not a blocker

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**